src/hipscat/io/file_io/file_pointer.py

```diff
@@ -1,16 +1,106 @@
+from typing import List, NewType
 import glob
 import os
-from typing import List, NewType
+import fsspec
+
+SUPPORTED_PROTOCOLS = ["file", "abfs", "s3"]
 
 FilePointer = NewType("FilePointer", str)
 """Unified type for references to files."""
 
 
-def get_file_pointer_from_path(path: str) -> FilePointer:
+def get_file_protocol(pointer: FilePointer) -> str:
+    """Method to parse filepointer for the filesystem protocol.
+        if it doesn't follow the pattern of protocol://pathway/to/file, then it 
+        assumes that it is a localfilesystem.
+    
+    Supported protocols: ["file", "abfs", "s3"]
+
+    Args:
+        pointer: filesystem pathway pointer
+    
+    Raises:
+        NotImplementedError: if protocol is not supported
+        NotImplementedError: if more than one protocol is in the FilePointer
+    """
+
+    if not isinstance(pointer, str):
+        pointer = str(pointer)
+
+    protocol = fsspec.utils.get_protocol(pointer)
+
+    if protocol not in SUPPORTED_PROTOCOLS:
+        raise NotImplementedError(f"{protocol} is not supported for hipscat!")
+    return protocol
+
+
+def get_fs(file_pointer: FilePointer, storage_options: dict = None) -> fsspec.filesystem:
+    """Create the abstract filesystem
+    
+    Args:
+        file_pointer: filesystem pathway
+        storage_options: dictionary that contains abstract filesystem credentials
+    """
+    if storage_options is None:
+        storage_options = {}
+    protocol = get_file_protocol(file_pointer)
+    file_pointer = get_file_pointer_for_fs(protocol, file_pointer)
+    return fsspec.filesystem(protocol, **storage_options), file_pointer
+
+
+def get_file_pointer_for_fs(protocol: str, file_pointer: FilePointer) -> FilePointer:
+    """Creates the filepathway from the file_pointer. Will strip the protocol so that
+            the file_pointer can be accessed from the filesystem
+        abfs filesystems DO NOT require the account_name in the pathway
+        s3 filesystems DO require the account_name/container name in the pathway
+
+    Args:
+        protocol: str filesytem protocol, file, abfs, or s3
+        file_pointer: filesystem pathway
+    
+    """
+
+    if not isinstance(file_pointer, str):
+        file_pointer = str(file_pointer)
+
+    if protocol == "file":
+        #return the entire filepath for local files
+        if "file://" in file_pointer:
+            split_pointer = file_pointer.split("file://")[1]
+        else:
+            split_pointer = file_pointer
+    if protocol == "abfs":
+        #return the path minus protocol+account name
+        split_pointer = file_pointer.split("abfs://")[1]
```

Comment:
How does this remove the account name?

Reply:
oops it doesn't. When I first was testing implementation for abfs, I thought that it would require the account name in the URI, but it doesn't. So everywhere I would just strip the account_name from the URI, before I realized that it just needs `abfs:///container/path/filename.ext` instead of `abfs:///account_name/container/path/filename.ext`. This is a remnant of that implementation. I've changed the method now